aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -57,21 +57,26 @@ Tensor& bitwise_not_out(Tensor& result, const Tensor& self) {
 }
 
 Tensor logical_not(const Tensor& self) {
-  Tensor result = at::empty({0}, self.options());
+  Tensor result = at::empty({0}, self.options().dtype(kBool));
   return at::logical_not_out(result, self);
 }
 
 Tensor& logical_not_(Tensor& self) {
-  return at::logical_not_out(self, self);
+  auto iter = TensorIterator::unary_op(self, self, /*check_internal_overlap=*/true);
+  logical_not_inplace_stub(iter.device_type(), iter);
+#ifdef BUILD_NAMEDTENSOR
+  at::namedinference::propagate_names(self, self);
+#endif
```

Comment:
@zou3519 I suppose `propagate_names` is not needed here. Is my assumption correct?

Reply:
Yeah, it's not necessary anymore because TensorIterator handles name propagation already.